client/internal/publicapi/publicapi.go

```diff
@@ -0,0 +1,79 @@
+package publicapi
+
+import (
+	"context"
+	"github.com/CiscoDevnet/terraform-provider-cdo/go-client/internal/http"
+	"github.com/CiscoDevnet/terraform-provider-cdo/go-client/internal/publicapi/transaction"
+	"github.com/CiscoDevnet/terraform-provider-cdo/go-client/internal/publicapi/transaction/transactionstatus"
+	"github.com/CiscoDevnet/terraform-provider-cdo/go-client/internal/retry"
+)
+
+func PostForTransaction(ctx context.Context, client http.Client, url string, body any, options retry.Options) (transaction.Type, error) {
+
+	req := client.NewPost(ctx, url, body)
+
+	t, err := sendAndCheckForError(req)
+	if err != nil {
+		return t, err
+	}
+	if isDone(t) {
+		return t, nil
+	}
+
+	return poll(ctx, client, options, t.TransactionPollingUrl)
```

Comment:
I would separate polling for a transaction to finish and triggering transaction into two separate methods so that it is clear that is what we do. I was very confused for a while.

Reply:
Make sense, done 👍